src/sqlfluff/core/rules/std/__init__.py

```diff
@@ -0,0 +1,20 @@
+# No docstring here as it would appear in the rules docs.
+# Rule definitions for the standard ruleset, dynamically imported from the directory.
+# noqa
+
+import os
+from importlib import import_module
+from glob import glob
+
+rules_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), "L*.py")
+
+for module in sorted(glob(rules_path)):
+    # Manipulate the module path to extract the filename without the .py
+    rule_id = os.path.splitext(os.path.basename(module))[0]
+    rule_class_name = f"Rule_{rule_id}"
+    rule_class = getattr(
+        import_module(f"sqlfluff.core.rules.std.{rule_id}"), rule_class_name
```

Comment:
I think depending on the filename *and* the classname here feels like overkill.

Based on the convo in #680 I think it would be better for the classname in each file to be consistent (maybe just `Rule()`?) and then here we just call `rule_class = import_module(f"sqlfluff.core.rules.std.{rule_id}").Rule`. This would mean that _recoding_ a rule just means renaming the file rather than renaming the file *and* the class. It also avoids the potential issue of someone having inconsistent naming between the file and the class.

Reply:
Maybe for double bonus points, we should wrap that line in a `try` clause, which catches the `ImportError` (or `AttributeError`?) in the case that no `Rule()` is defined in the file - and presents the user with a sensible error message?

Reply:
> Maybe for double bonus points, we should wrap that line in a try clause, which catches the ImportError (or AttributeError?) in the case that no Rule() is defined in the file - and presents the user with a sensible error message?

good idea - actioned